src/main/java/frc/robot/commands/tank/DriveTankCommand.java

```diff
@@ -7,41 +7,40 @@
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.tank.TankSubsystem;
 
-/** An example command that uses an example subsystem. */
+/** A command to drive the robot straight for x inches */
+// Just a rough implementation for testing;
+// TODO: make actually useful commands
 public class DriveTankCommand extends CommandBase {
   @SuppressWarnings({ "PMD.UnusedPrivateField", "PMD.SingularField" })
   private final TankSubsystem tankSubsystem;
 
-  private double yScale;
-  private double angularScale;
-  private boolean squareInput;
-
-  // TODO can we concentrate this into one
-  public DriveTankCommand(TankSubsystem tankSubsystem, double yScale, double angularScale) {
-    this(tankSubsystem, yScale, angularScale, true);
-  }
-
-  public DriveTankCommand(TankSubsystem tankSubsystem, double yScale, double angularScale, boolean squareInput) {
-    super();
+  private double distance;
 
+  public DriveTankCommand(TankSubsystem tankSubsystem, double inches) {
     this.tankSubsystem = tankSubsystem;
-
+    
     addRequirements(tankSubsystem);
 
-    this.yScale = yScale;
-    this.angularScale = angularScale;
-    this.squareInput = squareInput;
+    this.distance = inches;
   }
 
   // Called every time the scheduler runs while the command is scheduled.
   @Override
   public void initialize() {
-    tankSubsystem.setCarDrivePowers(yScale, angularScale, squareInput);
+    // TODO: the zeroing of the sensors takes a non-zero amount of time and `isFinished()` returns 
+    // true before the sensors are zeroed.
+    // Every other run of this command will fail, with the second run serving only to zero the sensors 
```

Comment:
can't we save the starting position of the encoders and simply add to it?

also, for serious movement, we should just take the position from the poseestimator instead of doing encoder logic ourselves. we can probably put .distanceToPoint(), etc. in a wrapper for all localization stuff

Reply:
You should consider a localization system for the future.